driven/storage/adapter.go

```diff
@@ -680,6 +681,15 @@ func (sa Adapter) GetMessages(orgID string, appID string, userID *string, messag
 	if len(messageIDs) > 0 {
 		filter = append(filter, primitive.E{Key: "_id", Value: bson.M{"$in": messageIDs}})
 	}
+
+	if read != nil {
+		filter = append(filter, primitive.E{Key: "recipients.read", Value: bson.M{"$ne": true}})
+	}
+
+	if mute != nil {
+		filter = append(filter, primitive.E{Key: "recipients.mute", Value: bson.M{"$ne": true}})
+	}
+
```

Comment:
Seems you didn't understood me well. 

Example:
if mute != nil {
		if *mute {
			filter = append(filter, primitive.E{Key: "recipients.mute", Value: true})	
		} else {
			// support of existing records where this field is missing
			filter = append(filter, primitive.E{Key: "recipients.mute", Value: bson.M{"$ne": true}}) 	
		}
	}